app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesActivity.java

```diff
@@ -121,34 +111,39 @@ public void checkInAlertBuilder(final int index) {
             + "Ticket: " + thisAttendee.getTicket().getType())
             .setPositiveButton("OK", (dialog, which) -> {
                 Log.d(TAG, "onClick: inside ok");
-                changeCheckStatus(thisAttendee.getId(), index);
+                attendeesPresenter.toggleAttendeeCheckStatus(index, thisAttendee.getId());
```

Comment:
Actually, you can't assume that network is reliable and simple sorting will definitely change the index .

Better would be to, just send the attendee to get updated and once it comes back updates, presenter searches the local list for the item with the same ID, and sends the updated list to view

This is definitely a little slower than previous method but is clean, reliable and has no dependency on the data managed by view